quickstarts/WFLY-15746_WildFly_Application_Development_Guide.adoc

```diff
@@ -0,0 +1,83 @@
+= WildFly Application Development Guide
+:author:            Eduardo Martins
+:email:             emartins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This document specifies the development of the Development Guide feature enhancement.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-15746[WFLY-15746]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1845[EAP7-1845]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:
+
+=== Affected Projects or Components
+
+WildFly (WFLY Documentation and Quickstarts Components)
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* The guide should have 2 major sections, one for first time/beginners app developers, and one for advanced app developers.
+
+* The major section for first time app developers should include the following new content:
+** How to develop a first basic application (helloworld quickstart) from scratch using Maven, a complete user story walkthrough, which should include:
```

Comment:
Do we think this is one quickstart or a small subset of quickstarts?  I don't think it would be a massive number but it feels like there may be a few getting started approaches.

Reply:
This is just the first app from scratch, in my mind just the simplest hello world

Reply:
This development guide should have some guiding principles that showcases efficient way to develop with WildFly.

While Jakarta EE development does not strictly adhere to https://12factor.net, there are principles that makes developing and operating WildFly much more simpler.

In particular, having the codebase being the source of deployments is a key aspect.
We should recommend & showcase how to provision and configure WildFly using the wildfly-maven-plugin in a reproducible & automated manner.
When we ask the user to copy a XML snippet in `standalone.xml`, we are not providing them a mechanism to keep this change in sync with WildFly releases and put the onus on them to automate this.
Compared to using a CLI scripts to apply this change to a provisioned WildFly server, this is automated, testable without any change to make outside of the codebase.

Reply:
IMHO we should include guidance for upstream only features, but consider those as alternatives to the supported features. If an upstream feature is a good enhancement than we should prioritise getting it supported.

Reply:
I'm not sure to follow you.
This guide must focus on the best way to develop with WildFly. If or when this will go in a supported product (I suppose you mean JBoss EAP) should not impact how WildFly is documented.

Reply:
The comment above is also a good opportunity to split this dev guides in different focused guides that can then be picked and choose when a feature reach a downstream product.

Reply:
I mean this is the upstream for an EAP 8 guide, it should prioritise now on the supported features.